controllers/factory.go

```diff
@@ -465,8 +465,16 @@ func (f *Factory) withCloudStackValidatorRegistry() *Factory {
 	return f
 }
 
+func (f *Factory) withCiliumTemplater() *Factory {
+	f.dependencyFactory.
+		WithHelmClientFactory(f.manager.GetClient()).
+		WithCiliumTemplater()
```

Comment:
i would prefer if we didn't trust the execution order to overwrite the default client factory

what we can do here is to have WithHelmClientFactory and WithCiliumTemplater defined in this controllers factory, so we can have our own logic